### ISSUE TITLE
Using cam without display?

### ISSUE THREAD

**Issue by dreiduke** [maxritter/DIY-Thermocam#115]
Waiting since a monat for the LCD,
Is it possible to use the cam without display?
When its connected to the computer, thermocam live viewer recognizes the cam.
But says, put it in the live mode.

I tried to push the button on the cam, short and long, no change.
Cannot press any button within live viewer.

**Comment by maxritter** [comment]
Hello @dreiduke,

yes this is possible. There is the checkNoDisplay function inside the diagnostics that runs after the device boots. If it detects that there is no LCD controller and no Touch controller attached, it will automatically switch into USB Serial Mode: https://github.com/maxritter/diy-thermocam/blob/679eee969cb9601244d535ed68b883560b834834/firmware/3.0/src/hardware/connection.cpp#L53

If you have VS Code installed on your PC, you can add some print statements to the function to see if it is executed: https://github.com/maxritter/diy-thermocam/tree/master/firmware

When the Device is in USB Serial Mode, it should be detected by the Thermal Live Viewer (https://github.com/maxritter/diy-thermocam/blob/master/software/thermal_live_viewer/README.MD)

Let me know if this helps!

**Comment by dreiduke** [comment]
Danke dir,

Und schöne restliche Weihnachten, und nen guten rutsch.

Werd das breakoutboard erstmal an nem Raspi probieren.
Denn der thermal live viewer, lässt mich nix drücken, solang die cam nicht in den serial mide geht, (ich mein den com10 belegt die Cam) Eventuelle print commands werd ich erst später probieren können, habe Dienst, über die Feiertage.

In diesem Sinne, und danke für die rückmeldung, die schnelle

Lg dreiduke alias Falko Wagner
Austria

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by androjet** [comment]
Hi!

Can the self-diagnostic results be viewed without a display? 

My device seems to be detected by the Thermal Live Viewer but I have no image. 

Thanks for your time!

**Comment by maxritter** [comment]
@androjet: Yes, it's also possible to fetch this information via Serial Connection: https://www.diy-thermocam.net/docs/serial/ (0x7F - getDiagnostic)

**Comment by androjet** [comment]
Thanks!

One more question if you have time. I'm new to coding and doing some troubleshooting.

I'm trying to input the diagnostic codes through the serial monitor-

With IDE: I can connect to the teensy via serial, but get no response from the teensy on any baud rate

VS Code: Serial monitor opens but appears to be constantly refreshing and I cant enter send anything to the teensy

I'm able to communicate when I upload my own test firmware, but not the thermocam firmware. Is there anything unique about the thermocam serial communication that I'm missing?

Thanks for your time and patience!

**Comment by maxritter** [comment]
@androjet Nothing special about the serial communication, it inits at 115200 baudrate (https://github.com/maxritter/diy-thermocam/blob/771a51fd7ccbe0f914daafdd735f9575367b5861/firmware/3.0/src/hardware/hardware.cpp#L535) and then you will be able to send the commands from the serial protocol and/or receive Serial.println() commands.

When I was on Windows I always used the application hTerm to debug Serial stuff, this one worked for sure ;)

**Comment by androjet** [comment]
Got it! Commenting for anyone else who has this issue:

I may have been doing this the hard way, but it seemed my camera was getting hung up on some code:

-A few while loops waiting for the for a display touch (which didn't happen because I have no display connected)
-live mode helper (no idea why that was an issue)

I was able to comment out those lines and get it working. 

Works great with the **Thermal Live Viewer.** 

Unfortunately I wasn't able to get it to work with the **Thermal Data Viewer** but could be user error.

**Comment by androjet** [comment]
> Waiting since a monat for the LCD, Is it possible to use the cam without display? When its connected to the computer, thermocam live viewer recognizes the cam. But says, put it in the live mode.
> 
> I tried to push the button on the cam, short and long, no change. Cannot press any button within live viewer.

I had the exact issue. I was getting stuck at "put camera in live mode!" because the flir wasnt connecting properly.

In addition to commenting out some of the display while() loops, it was really helpful to add a while() loop with a serial print command that shows the lepton error code. You can find all that in the check hardware functions. My lepton board was very finicky and this allowed me to fix the connection with live feedback.

Once I got the error code to return as "0" consistently, the live viewer properly ran the diagnostic and displayed the flir image.
